tests/PHPStan/Rules/Comparison/data/bug-4371.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Bug4371;
+
+class Foo {
+}
+
+class Bar extends Foo {
+
+}
+
+if(is_a(Bar::class, Foo::class)) { // should be reported
+	echo "This will never be true";
+} else {
+	echo "NO";
+}
+
+if(is_a(Bar::class, Foo::class, false)) { // should be reported
```

Comment:
this is interessting.. 

the test-case works for line 12 and 18 when testing them in isolation.
having both of them in a single file - as the test looks right now - does not report a error on line 18 because the expression is considered already specified, and [therefore early returns](https://github.com/phpstan/phpstan-src/blob/69fd2bc8504ad2fed03bc1f1377fb34bfb6b5964/src/Rules/Comparison/ImpossibleCheckTypeHelper.php#L194-L196).

I think this is a unrelated bug. should I separate the test into 2 files, so we can proceed with this PR?

Reply:
> should I separate the test into 2 files,

You don't need 2 files, just isolate the scope by putting the code into two separate anonymous functions.